### PR TITLE
Migration base module

### PR DESCRIPTION
**Pull request by alexandraorth**

### CODE REVIEW

**Comment by fppt** [comment]
NO ONE MERGE THIS PLEASE. Just checked with @aelred and we not 100% sure it is safe to merge. Best wait until @alexandraorth gets back.

**Comment by alexandraorth** [comment]
As long as this passed jenkins, it is safe to merge. If the release hasn't done out yet, can we add this in?

File: mindmaps-migration/base/src/main/java/io/mindmaps/migration/base/AbstractMigrator.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+    /**
+     * Partition a stream into a stream of collections, each with batchSize elements.
+     * @param iterator Iterator to partition
+     * @param <T> Type of values of iterator
+     * @return Stream over a collection that are each of batchSize
+     */
+    protected <T> Stream<Collection<T>> partitionedStream(Iterator<T> iterator){
+        return StreamSupport.stream(Spliterators.spliteratorUnknownSize(
+                Iterators.partition(iterator, batchSize), Spliterator.ORDERED), false);
```
Comment (aelred) [inline mindmaps-migration/base/src/main/java/io/mindmaps/migration/base/AbstractMigrator.java]: Beautiful stream magic :)

File: mindmaps-migration/base/pom.xml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>migration-base</artifactId>
+    <parent>
+        <artifactId>mindmaps-migration</artifactId>
+        <groupId>io.mindmaps</groupId>
+        <version>0.5.0-SNAPSHOT</version>
```
Comment (aelred) [inline mindmaps-migration/base/pom.xml]: I'll pull this down and bump up the version

File: mindmaps-migration/base/src/test/java/io/mindmaps/migration/base/MigratorTestUtil.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package io.mindmaps.migration.base;
+
+/**
+ * Created by alexandra on 10/20/16.
+ */
+public class MigratorTestUtil {
+}
```
Comment (aelred) [inline mindmaps-migration/base/src/test/java/io/mindmaps/migration/base/MigratorTestUtil.java]: Whoops

**Comment by aelred** [comment]
please retest this please :/

**Comment by haikalpribadi** [comment]
Can one of the admins verify this patch?

**Comment by aelred** [comment]
please retest this please

**Comment by aelred** [comment]
please retest this please

**Pull request merged**